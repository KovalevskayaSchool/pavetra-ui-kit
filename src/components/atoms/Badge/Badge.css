.ks-badge {
  color: inherit;
  position: relative;
  display: inline-block;
  width: fit-content;
}

.ks-badge__counter {
  display: flex;
  justify-content: center;
  align-items: center;

  overflow: hidden;
  position: absolute;
  top: 0;
  inset-inline-end: 0;
  transform: translate(50%, -50%);
  transform-origin: 100% 0%;

  background-color: var(--color-danger);
  border-radius: var(--border-radius-rounded);
  box-shadow: 0 0 0 1px #fff;
  transition: background 0.2s;

  white-space: nowrap;
  text-align: center;
  z-index: auto;
  min-width: 20px;
  height: 20px;
}

.ks-badge__label {
  color: #fff;
  font-weight: 500;
  font-size: 12px;
  line-height: 20px;
  white-space: nowrap;
  text-align: center;
}
