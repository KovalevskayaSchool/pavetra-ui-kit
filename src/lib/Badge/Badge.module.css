.badge {
  color: inherit;
  position: relative;
  display: inline-block;
  width: fit-content;

  --badge-fg-color: var(--color-white);
  --badge-bg-color: var(--color-danger);
}

.badge__counter {
  display: flex;
  justify-content: center;
  align-items: center;

  overflow: hidden;
  position: absolute;
  inset-block-start: 0.25rem;
  inset-inline-end: 0;
  transform: translate(50%, -50%);
  transform-origin: 100% 0%;

  background-color: var(--badge-bg-color);
  border-radius: var(--border-radius-rounded);
  box-shadow: 0 0 0 1px var(--badge-fg-color);
  transition: all var(--transition-duration) ease-out;

  white-space: nowrap;
  text-align: center;
  z-index: 1;
  min-width: 1.25rem;
  height: 1.25rem;
}

.badge__label {
  color: var(--badge-fg-color);
  font-weight: 500;
  font-size: 0.7rem;
  line-height: 1.25rem;
  white-space: nowrap;
  text-align: center;
}
