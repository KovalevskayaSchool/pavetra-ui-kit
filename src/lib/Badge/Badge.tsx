import { forwardRef, PropsWithChildren } from "react";
import cn from "classnames";
import styles from "./Badge.module.css";

export interface BadgeProps {
  className?: string;
  count?: number;
}

export const Badge = forwardRef<HTMLSpanElement, PropsWithChildren<BadgeProps>>(
  ({ className, count = 0, children, ...props }, ref) => {
    return (
      <span ref={ref} className={cn(className, styles["badge"])} {...props}>
        {count > 0 && (
          <sup className={styles["badge__counter"]}>
            <span className={styles["badge__label"]}>{count.toString()}</span>
          </sup>
        )}
        {children}
      </span>
    );
  }
);
