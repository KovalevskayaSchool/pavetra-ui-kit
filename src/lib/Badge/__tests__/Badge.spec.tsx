import React from 'react'
import { render } from "@testing-library/react";

import { Badge } from "../Badge";

describe("Badge", () => {
  test("should render component", () => {
    const { container } = render(<Badge />);

    expect(container).toMatchSnapshot();
  });
});
