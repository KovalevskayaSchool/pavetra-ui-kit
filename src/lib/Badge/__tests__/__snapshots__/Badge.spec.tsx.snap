// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render component 1`] = `
<div>
  <span
    class="badge"
  />
</div>
`;
